File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/viewmodels/GroupFragmentViewModel.java
package vng.zalo.tdtai.zalo.zalo.viewmodels;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import com.google.firebase.firestore.DocumentSnapshot;
import java.util.ArrayList;
import java.util.List;
import vng.zalo.tdtai.zalo.zalo.ZaloApplication;
import vng.zalo.tdtai.zalo.zalo.models.RoomItem;
import vng.zalo.tdtai.zalo.zalo.utils.Constants;

public class GroupFragmentViewModel extends ViewModel {
    private static final String TAG = GroupFragmentViewModel.class.getSimpleName();

    public MutableLiveData<List<RoomItem>> liveRoomItems;

    public GroupFragmentViewModel() {
        liveRoomItems = new MutableLiveData<>(new ArrayList<>());

        Log.d(TAG, ZaloApplication.currentUser.toString());

        ZaloApplication.getFirebaseInstance()
                .collection(Constants.COLLECTION_USERS)
                .document(ZaloApplication.currentUser.phone)
                .collection(Constants.COLLECTION_ROOMS)
                .whereEqualTo("roomType", Constants.ROOM_TYPE_GROUP)
                .orderBy("lastMsgTime")
                .addSnapshotListener((queryDocumentSnapshots, e) -> {
                    if (queryDocumentSnapshots != null){
                        List<RoomItem> roomItems = new ArrayList<>();
                        for (DocumentSnapshot doc : queryDocumentSnapshots) {
                            RoomItem roomItem = doc.toObject(RoomItem.class);
                            assert roomItem != null;
                            roomItem.roomId = doc.getId();
                            roomItems.add(roomItem);
                        }
                        liveRoomItems.setValue(roomItems);
                        Log.d(TAG, liveRoomItems.getValue().toString());
                    } else {
                        Log.d(TAG, Constants.COLLECTION_USERS+"/"+ZaloApplication.currentUser.phone+"/"+Constants.COLLECTION_ROOMS);
                        Log.d(TAG, "queryDocumentSnapshots is null");
                    }
                });
    }
}

File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/viewmodels/GroupFragmentViewModel.kt
package vng.zalo.tdtai.zalo.zalo.viewmodels

import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import vng.zalo.tdtai.zalo.zalo.ZaloApplication
import vng.zalo.tdtai.zalo.zalo.models.RoomItem
import vng.zalo.tdtai.zalo.zalo.utils.Constants
import java.util.*

class GroupFragmentViewModel : ViewModel() {

    val liveRoomItems: MutableLiveData<List<RoomItem>>

    init {
        liveRoomItems = MutableLiveData(ArrayList())

        Log.d(TAG, ZaloApplication.currentUser!!.toString())

        ZaloApplication.firebaseInstance
                .collection(Constants.COLLECTION_USERS)
                .document(ZaloApplication.currentUser!!.phone!!)
                .collection(Constants.COLLECTION_ROOMS)
                .whereEqualTo("roomType", Constants.ROOM_TYPE_GROUP)
                .orderBy("lastMsgTime")
                .addSnapshotListener { queryDocumentSnapshots, _ ->
                    if (queryDocumentSnapshots != null) {
                        val roomItems = ArrayList<RoomItem>()
                        for (doc in queryDocumentSnapshots) {
                            roomItems.add(
                                    doc.toObject(RoomItem::class.java).apply {
                                        roomId = doc.id
                                    }
                            )
                        }
                        liveRoomItems.value = roomItems
                        Log.d(TAG, liveRoomItems.value!!.toString())
                    } else {
                        Log.d(TAG, Constants.COLLECTION_USERS + "/" + ZaloApplication.currentUser!!.phone + "/" + Constants.COLLECTION_ROOMS)
                        Log.d(TAG, "queryDocumentSnapshots is null")
                    }
                }
    }

    companion object {
        private val TAG = GroupFragmentViewModel::class.java.simpleName
    }
}